aiida/cmdline/commands/cmd_process.py

```diff
@@ -185,11 +185,33 @@ def process_call_root(processes):
 @click.option(
     '-m', '--max-depth', 'max_depth', type=int, default=None, help='Limit the number of levels to be printed.'
 )
+@click.option(
+    '-e', '--end', 'last', is_flag=True, default=False, help='Report for the most recent node in verdi process list -a'
+)
 @decorators.with_dbenv()
-def process_report(processes, levelname, indent_size, max_depth):
+def process_report(processes, levelname, indent_size, max_depth, last):
     """Show the log report for one or multiple processes."""
     from aiida.cmdline.utils.common import get_calcjob_report, get_process_function_report, get_workchain_report
-    from aiida.orm import CalcFunctionNode, CalcJobNode, WorkChainNode, WorkFunctionNode
+    from aiida.orm import CalcFunctionNode, CalcJobNode, WorkChainNode, WorkFunctionNode, load_node
+    from aiida.tools.query.calculation import CalculationQueryBuilder
+
+    if last:
+        # query all nodes, getting the last one available
+        builder = CalculationQueryBuilder()
+        lastnode_info = builder.get_projected(builder.get_query_set(), CalculationQueryBuilder.default_projections)[-1]
+        # take pk from node info and load that node
+        pk = lastnode_info[0]
+        node = load_node(pk)
+        # set process list to the singular final node
+        processes = [node]
```

Comment:
I wouldn't use the `CalculationQueryBuilder` here since that is more complicated than we need. We can use a simple `QueryBuilder` directly:
```suggestion
    if most_recent_node:
        builder =  QueryBuilder().append(ProcessNode, tag='n').order_by({'n': {'ctime': 'desc'}})
        processes = builder.first(flat=True)
```

Reply:
Perfect, this seems to be a much more efficient implementation. I am still quite unfamiliar with the `QueryBuilder` family (clearly!)

Reply:
The `CalculationQueryBuilder` is simply a utility class that builds on top of the `QueryBuilder` functionality. It is mostly designed and used for the `verdi process list` command. If you need a simple query, your best bet is to go straight to the `QueryBuilder` yourself